charts/kong/values.yaml

```
@@ -358,7 +358,7 @@ ingressController:
     # The annotations for service account
     annotations: {}
 
-  installCRDs: true
```

After this PR, `installCRDs` is very very much of a misnomer for this setting (because it doesn't directly lead to CRDs being installed or not). Can you think of a way to make it clear in all possible places that this setting only affects the legacy way of installing CRDs? Maybe it would be justified to rename this setting (e.g. `deprecatedInstallCRDs` or `deprecatedInstallCRDsWithinRelease`) and fail loudly if `installCRDs` (the old setting) is set in the user's values without change, thus forcing users to carefully consider the change of semantics?

---

It still does install managed CRDs. The annoying thing with this migration is that there's no good way to ever migrate off it automatically: if you'd ever used the old CRD installation method in the past, you have to continue using it. We can't really discard the old system entirely, so it does still work.

Per the original PR description, Helm doesn't provide a way to remove managed resources from a release. They're still part of it even if you remove the `app.kubernetes.io/managed-by: Helm` label and `meta.helm.sh/release-name` annotation, since Helm tracks them in its release Secret (which isn't designed to allow manual edits) and will delete resources that are there but no longer in template output. There's no good migration path off the legacy system, so I don't think we can deprecate it, even though we don't want any new releases to use it.

We can hide this setting entirely, however, as legacy behavior detection doesn't rely on it being present. Hidden settings are atypical, but this seems like a justifiable case to use them.

---

bf98a5cba hides the setting with the expectation that nobody will explicitly set it going forward, which aligns us with Helm's upstream stance on CRD management. Documentation now mentions the automatic mechanism and how to check for release-managed CRDs.

There's still no warning for the legacy behavior because of the lack of a good migration path for existing releases that use it, but we no longer expose the option at all for new releases.

Also added the terminology recommendations from other comments alongside that with some changes (just using the standard Helm terms only).